Library/Homebrew/utils/shared_audits.rb

```diff
@@ -87,30 +77,33 @@ def gitlab_release_data(user, repo, tag)
     end
   end
 
-  GITLAB_PRERELEASE_ALLOWLIST = {}.freeze
-
-  def gitlab_release(user, repo, tag, formula: nil)
+  def gitlab_release(user, repo, tag, formula: nil, cask: nil)
     release = gitlab_release_data(user, repo, tag)
     return unless release
 
     return if Date.parse(release["released_at"]) <= Date.today
-    return if formula && GITLAB_PRERELEASE_ALLOWLIST[formula.name] == formula.version
+
+    exception = if formula
+      tap_audit_exception(:gitlab_prerelease_allowlist, formula.tap, formula.name)
+    elsif cask
+      tap_audit_exception(:gitlab_prerelease_allowlist, cask.tap, cask.token)
+    end
+    version = if formula
+      formula.version
+    elsif cask
+      cask.version
+    end
```

Comment:
Might be nice to set `exception` and `version` in one place so there's a single `if`